hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/events/EventQueue.java

```diff
@@ -205,11 +192,14 @@ public <PAYLOAD, EVENT_TYPE extends Event<PAYLOAD>> void fireEvent(
         for (EventHandler handler : executorAndHandlers.getValue()) {
           queuedCount.incrementAndGet();
           if (LOG.isTraceEnabled()) {
-            LOG.trace(
-                "Delivering [event={}] to executor/handler {}: <json>{}</json>",
-                event.getName(),
-                executorAndHandlers.getKey().getName(),
-                TRACING_SERIALIZER.toJson(payload).replaceAll("\n", "\\\\n"));
+            String jsonPayload = serializeObject(payload);
+            if (jsonPayload != null) {
+              LOG.trace(
+                  "Delivering [event={}] to executor/handler {}: <json>{}</json>",
+                  event.getName(),
+                  executorAndHandlers.getKey().getName(),
+                  jsonPayload.replaceAll("\n", "\\\\n"));
+            }
```

Comment:
If serialization fails, let's still log it without the payload, at debug level (using the `else if` branch).

Reply:
If serialization fails, we now log at debug level without payload